src/Libraries/DSOffice/Excel.cs

```diff
@@ -406,25 +413,30 @@ private static object[][] ConvertToJaggedArray(object[,] input)
         /// <summary>
         /// return data from given worksheet (GetDataFromExcelWorksheet node)
         /// </summary>
-        internal object[][] Data 
-        { 
+        internal object[][] Data
+        {
             get
             {
-                var vals = ws.UsedRange.get_Value();
+                return DataAsStrings();
+            }
+        }
 
-                // if worksheet is empty
-                if (vals == null)
-                    return new object[0][];
+        internal object[][] DataAsStrings(bool convertToString = false)
```

Comment:
It is a bit weird to have a function name which means it will return strings, but the parameter can decide whether to convert to strings.